modules/engines/engine-hdb/src/main/java/com/sap/xsk/hdb/ds/transformer/hdbdd/HdbddTransformer.java

```diff
@@ -114,8 +126,52 @@ public XSKDataStructureHDBTableModel transformEntitySymbolToTableModel(EntitySym
 
     tableModel.setColumns(tableColumns);
     tableModel.setLocation(location);
-    if (entitySymbol.getAnnotation(CATALOG_ANNOTATION) != null) {
+    if (entitySymbol.getAnnotation(CATALOG_ANNOTATION) != null){
       tableModel.setTableType(entitySymbol.getAnnotation(CATALOG_ANNOTATION).getKeyValuePairs().get(CATALOG_OBJ_TABLE_TYPE).getValue());
+
+      if (entitySymbol.getAnnotation(CATALOG_ANNOTATION).getKeyValuePairs().get(INDEX) != null){
+        List<XSKDataStructureHDBTableIndexModel> indexes = new ArrayList<>();
+        List<XSKDataStructureHDBTableConstraintUniqueModel> uniqueIndexes = new ArrayList<>();
+        AnnotationArray catalogIndexAnnArray = (AnnotationArray) entitySymbol.getAnnotation(CATALOG_ANNOTATION).getKeyValuePairs().get(INDEX);
+
+        for (AbstractAnnotationValue currentAnnValue : catalogIndexAnnArray.getValues()){
+          AnnotationObj annObject = (AnnotationObj) currentAnnValue;
```

Comment:
Do not shorten variable names, use `annotationObject`, `currentAnnotationValue`, etc.